lib/Backends/Interpreter/InterpreterNodes.cpp

```diff
@@ -1361,25 +1361,140 @@ void BoundInterpreterFunction::fwdGatherRangesInst(
   }
 }
 
-void BoundInterpreterFunction::fwdScatterAssignInst(
-    const glow::ScatterAssignInst *I) {
+template <typename ElemTy>
+void BoundInterpreterFunction::fwdScatterDataInstCopyImpl(
+    const glow::ScatterDataInst *I) {
   Tensor *dataT = getTensor(I->getData());
   Tensor *indicesT = getTensor(I->getIndices());
   Tensor *slicesT = getTensor(I->getSlices());
 
-  size_t dataSliceSize =
-      dataT->size() / dataT->dims()[0] * dataT->getType().getElementSize();
+  const size_t numSlices = slicesT->size() / slicesT->dims()[0];
 
+  auto IH = indicesT->getHandle<int64_t>();
   // For each index, copy from the slice at that index into the location in data
   // given the offset from the indices tensor.
-  for (size_t i = 0, end = indicesT->size(); i < end; i++) {
-    size_t destDataIdx = indicesT->getHandle<int64_t>().raw(i);
+  for (size_t i = 0, end = indicesT->dims()[0]; i < end; i++) {
+    size_t destDataIdx = 0;
+    if (indicesT->dims().size() == 1) {
```

Comment:
Can you add some comments in here about what's going on?

Reply:
Hm, I think we can simplify it by always turning indices to 2-dimensional.